crates/astria-sequencer-client/src/lib.rs

```diff
@@ -311,33 +254,44 @@ pub trait SequencerClientExt: Client {
     /// # Errors
     ///
     /// - If calling tendermint `abci_query` RPC fails.
-    /// - If the opaque bytes cannot be deserialized as a sequencer query response.
-    /// - If the deserialized sequencer query response is not a balance response.
-    async fn get_nonce(&self, address: &Address, height: Option<Height>) -> Result<Nonce, Error> {
-        use borsh::BorshDeserialize as _;
-        use sequencer::accounts::query::Response;
+    /// - If the bytes contained in the abci query response cannot be read as an
+    ///   `astria.sequencer.v1alpha1.NonceResponse`.
+    async fn get_nonce(&self, address: [u8; 20], height: u32) -> Result<NonceResponse, Error> {
+        use proto::Message as _;
+
+        let prefix = b"accounts/nonce/";
+        let mut path = vec![0u8; prefix.len() + 20 * 2];
+        path.extend_from_slice(prefix);
+        faster_hex::hex_encode(&address, &mut path).expect(
+            "this is a bug: the buffer holding the abci query path should have the correct \
+             precalculated size",
+        );
```

Comment:
same comments about the path as above

Reply:
ty for noticing this! see the other comment re fix